File: src/sort_algorithms/selection_sort/selection_sort.c
#include <stdio.h>

void swap(int *first, int *second){
    int temp;

    temp = *first;
    *first = *second;
    *second = temp;
}

void selection_sort(int *array, int len){
    int i, j, min, temp;

    for (i=0; i < len-1; i++){
        min = i;

        for (j=i+1; j < len; j++){
            if (array[j] < array[min]){
                min = j;
            }
        }
        swap(&array[i], &array[min]);
    }
}

void print_array(int *array, int len){
    for (int i=0; i<len; i++){
        printf("%d", array[i]);
        if (i < len-1){
            printf(", ");
        }
        else{
            printf("\n");
        }
    }
}

void main(){
    int array[10] = {213, 2, 54, 12, 53, 2, 75, 4, 674, 24};

    print_array(array, 10);

    selection_sort(array, 10);

    print_array(array, 10);
}

File: src/sort_algorithms/selection_sort/SelectionSort.java
import java.util.*;

public class SelectionSort{
    public static void main(String[] args){
        int array[] = {213, 2, 54, 12, 53, 2, 75, 4, 674, 24};

        print_array(array);

        array = selection_sort(array);

        print_array(array);
    }

    public static void print_array(int[] array){
        int len = array.length;

        for (int i=0; i<len; i++){
            System.out.printf("%d", array[i]);

            if (i < len-1){
                System.out.print(", ");
            }
            else {
                System.out.print("\n");
            }
        }
    }

    public static int[] selection_sort(int[] array){
        int len = array.length;

        for (int i=0; i<len-1; i++){
            int min = i;

            for (int j=i+1; j<len; j++){
                if (array[min] > array[j]){
                    min = j;
                }
            }
            int temp = array[i];
            array[i] = array[min];
            array[min] = temp;
        }

        return array;
    }
}